### PR TITLE
Sign in modal with terms and conditions

### PR DESCRIPTION
**Pull request by PopDaph**
Implement the modal to Sign in with terms and conditions, as defined here: https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=1845%3A45444&mode=design&t=FH3HOsS5pYng6yFi-1

Looks like this: 

Capture d’écran 2023-11-06 à 19 14 45

### CODE REVIEW

File: front/components/Button.tsx [hunk 1]
```diff
@@ -5,30 +5,10 @@ import {
   GoogleLogo,
   Icon,
   LoginIcon,
+  Modal,
   RocketIcon,
 } from "@dust-tt/sparkle";
-
-import { classNames } from "@app/lib/utils";
-
-export function GoogleSignInButton({
```
Comment (PopDaph) [inline front/components/Button.tsx:13]: This one was unused to so I removed the component.

**Review by spolu — COMMENTED** [review]
I'm not sure I perfectly understand the change. This is for all sign-in? Where is github?

**Review by spolu — COMMENTED** [review]
OK I see now, this is for Start With Dust and we push them to google :+1:

File: front/components/Button.tsx [hunk 2]
```diff
@@ -117,28 +97,55 @@ export function SignUpDropDownButton({
...
-          />
+      </div>
+
+      <Modal
+        isOpen={showModal}
+        onClose={() => setShowModal(false)}
+        hasChanged={false}
+        title="Start with Dust"
+        type="right-side"
+        width="wide"
```
Comment (spolu) [inline front/components/Button.tsx:120]: Should we make it normal? not wide?

**Review by spolu — DISMISSED** [review]
LGTM

The modal looks quite wide given how empty it is? otherwise :+1:

**Comment by PopDaph** [comment]
Yes, this is for Sign in - The sign up stays the same:
Capture d’écran 2023-11-07 à 09 59 44

File: front/components/Button.tsx (hunk 2 above)
Comment (PopDaph) [reply front/components/Button.tsx]: Done (checked with Ed 👍🏻 )

**Pull request merged**